src/Services/CheckoutService.php

```diff
@@ -1,251 +0,0 @@
-        if (is_a($deduction, app('discount'))) {
-            $cart->applyDiscount($deduction);
-        }
-
-        if (is_a($deduction, app('voucher'))) {
-            $cart->applyVoucher($deduction);
-        }
-
-        return $cart;
-    }
-
-    /**
-     * Apply discount to cart.
-     *
-     * @param Model $discount
-     * @param Cart $cart
-     * @return Cart
-     */
-    public function applyDiscountToCart($discount, $cart)
-    {
-        if (! in_array($discount->applies_to, [self::APPLICATION_ORDER, self::APPLICATION_PARTICIPANT])) {
-            return $cart;
-        }
-
-        $cart->discounts()->syncWithoutDetaching([$discount->id]);
-        $cart->total_deductions = $this->calculateCartDeductions($cart);
-        $cart->save();
-
-        return $cart;
-    }
-
-    /**
-     * Issue partial redemption voucher when total discounts are higher than total amount.
-     *
-     * @param  Cart $cart
-     * @return Model
-     */
-    public function issueCartPartialRedemptionVoucher($cart)
-    {
-        $order = $cart->order;
-
-        /** @psalm-suppress UndefinedMethod */
-        $voucher = app('voucher')::create([
-            'location_id' => $cart->location_id,
-            'customer_id' => $cart->user_id,
-            'voucher_type_id' => Cart::PARTIAL_REDEMPTION_VOUCHER_TYPE_ID,
-            'redeemable_at' => now(),
-            'amount' => $cart->total_deductions - ($cart->amount + $cart->total_taxes + $cart->total_fees),
-            'expires_at' => $cart->vouchers()->first()->expires_at,
-            'creator_id' => $cart->user_id,
-            'updater_id' => $cart->user_id,
-        ]);
-
-        $order->partial_redemption_voucher_id = $voucher->id;
-        $order->save();
-
-        return $voucher;
-    }
-
-    /**
-     * Calculate all cart deductions including items deductions.
-     *
-     * @return int
-     */
-    public function calculateCartDeductions($cart)
-    {
-        $totalDeductions = 0;
-
-        foreach ($cart->cartItems as $cartItem) {
-            $totalDeductions += $cartItem->total_deductions;
-        }
-
-        $totalDeductions += $this->calculateCartVouchers($cart);
-        $totalDeductions += $this->calculateCartDiscounts($cart);
-
-        return $totalDeductions;
-    }
-
-    /**
-     * Calculate total amount of cart vouchers (not counting cart items).
-     *
-     * @param Cart $cart
-     * @return int
-     */
-    public function calculateCartVouchers($cart)
-    {
-        $total = 0;
-
-        foreach ($cart->vouchers()->get() as $voucher) {
-            if ($voucher->amount > 0) {
-                $total += $voucher->amount;
-            }
-        }
-
-        return $total;
-    }
-
-    /**
-     * Calculate total amount of cart discounts (not counting cart items).
-     *
-     * @param Cart $cart
-     * @return int
-     */
-    public function calculateCartDiscounts($cart)
```

Comment:
Moved to CartDeduction::calculateCartDeduction - this version will be implemented in Discount package.